advanced-solidus/images-and-image-processing.md

```diff
@@ -0,0 +1,196 @@
+# Images and Image Processing
+
+Solidus utilizes [MiniMagick](https://github.com/minimagick/minimagick) for [ActiveStorage](https://edgeguides.rubyonrails.org/active_storage_overview.html) and [ImageMagick](https://imagemagick.org/index.php) for [Paperclip](https://github.com/thoughtbot/paperclip#paperclip) to handle the under-the-hood resizing and image processing. After an image is uploaded to a project, it is automatically resized based on when the view requests an image variant "style". The main dictating functions can be found in [adapter](https://github.com/solidusio/solidus/blob/master/core/app/models/concerns/spree/active_storage_adapter/attachment.rb) that controls sizing and the [ActiveStorage](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/image/active_storage_attachment.rb)/[Paperclip](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/image/paperclip_attachment.rb) attachment modules. Solidus is currently in the process of depreciating Paperclip due to ActiveStorage's inclusion with rails.
+
+## Customizing Size Defaults and Attachment Module Override
+
+As you can see from the linked modules above, the default configuration provides only a few variant sizes out of the box. Changing the image sizing is simply a configuration of an attachment module and is referenced when the attachment class method `style_to_size` is called. We could create ourselves a new adapter for our app and override the settings that way, but in this case, we just want to add some new types of sizings for our UI display.
+
+If you are running Solidus 3.1 or later, The most simple and most straight forward way to modify these settings is to append the configuration in our spree initializer.
+
+{% code title="config/initializers/spree.rb" %}
+```ruby
+# ...
+Spree::Config.image_sizes = {
+  mini: '48x48>',
+  small: '100x100>',
+  product: '240x240>',
+  large: '600x600>',
+  jumbo: '1600x1600>'
+}
+
+Spree::Config.image_default = :product
+```
+{% endcode %}
+
+This will overwrite the default configuration and add Amazing Store's custom sizing to the `:styles` hash.
+
+If you are utilizing versions of solidus that are 3.0 or before, we will have to insert an override to change our defaults or utilize our own ActiveStorageAttachment module. Inserting an override is relatively easy. The default pathing for override files is `app/overrides/`  (this can be changed within your application's `config/application.rb` file should you choose to do so), so we will be appending our `image_style` override there:
```

Comment:
> Inserting an override is relatively easy. The default pathing for override files is `app/overrides/`  (this can be changed within your application's `config/application.rb` file should you choose to do so), so we will be appending our `image_style` override there:

We have a page in the guides for this, let's link to it instead of re-explaining this again. WDYT?

Reply:
100% onboard with this, I felt it was a bit redundant

From the learning examples found in the [customizing the core guide](customization/customizing-the-core.md#L232), we can append our `image_style` override like: